packages/xstate-wallet/src/store/dexie-backend.ts

```diff
@@ -0,0 +1,203 @@
+
+  // Generic Getters
+
+  public async channels() {
+    return this.getAll(ObjectStores.channels);
+  }
+
+  public async objectives() {
+    return this.getAll(ObjectStores.objectives, true);
+  }
+  public async nonces() {
+    const nonces = await this.getAll(ObjectStores.nonces);
+    for (const key in nonces) {
+      if (nonces[key]) {
+        nonces[key] = new BigNumber(-1);
+      } else {
+        nonces[key] = new BigNumber(nonces[key]);
+      }
+    }
+    return nonces;
+  }
+
+  public async privateKeys() {
+    return this.getAll(ObjectStores.privateKeys);
+  }
+
+  public async ledgers() {
+    return this.getAll(ObjectStores.ledgers);
+  }
+
+  // Individual Getters
+  public async getBudget(key: string): Promise<SiteBudget | undefined> {
+    const budget: SiteBudget | undefined = await this.get(ObjectStores.budgets, key);
+    if (!budget) return budget;
+
+    return {
+      ...budget,
+      forAsset: _.mapValues(budget.forAsset, (assetBudget: AssetBudget) => ({
+        assetHolderAddress: assetBudget.assetHolderAddress,
+        availableReceiveCapacity: bigNumberify(assetBudget.availableReceiveCapacity),
+        availableSendCapacity: bigNumberify(assetBudget.availableSendCapacity),
+        channels: assetBudget.channels
+      }))
+    };
+  }
+
+  public async setBudget(key: string, value: SiteBudget) {
+    return this.put(ObjectStores.budgets, value, key);
+  }
+
+  public async deleteBudget(key: string) {
+    return this.delete(ObjectStores.budgets, key);
+  }
+
+  public async getChannel(key: string) {
+    // TODO: This is typed to return ChannelStoredData, but it actually
+    // returns ChannelStoreEntry.
+    // This happens all over the place.
+    const channel = await this.get(ObjectStores.channels, key);
+    return channel && ChannelStoreEntry.fromJson(channel);
+  }
+  public async getObjective(key: number) {
+    return this.get(ObjectStores.objectives, key);
+  }
+  public async getNonce(key: string) {
+    const nonce = await this.get(ObjectStores.nonces, key);
+    if (!nonce) {
+      return new BigNumber(-1);
+    }
+    return new BigNumber(nonce);
+  }
+  public async getPrivateKey(key: string) {
+    return this.get(ObjectStores.privateKeys, key);
+  }
+  public async getLedger(key: string) {
+    return this.get(ObjectStores.ledgers, key);
+  }
+
+  // Individual Setters
+
+  public async setPrivateKey(signingAddress: string, privateKey: string) {
+    const pksPutted = await this.put(ObjectStores.privateKeys, privateKey, signingAddress);
+    return pksPutted;
+  }
+  public async setChannel(key: string, value: ChannelStoredData) {
+    return this.put(ObjectStores.channels, value, key);
+  }
+  public async setLedger(key: string, value: string) {
+    return this.put(ObjectStores.ledgers, value, key);
+  }
+  public async setNonce(key: string, value: BigNumber) {
+    await this.put(ObjectStores.nonces, value.toString(), key);
+
+    return await this._db[ObjectStores.nonces].get(key);
```

Comment:
I think this is no longer necessary because in [here](https://github.com/statechannels/monorepo/pull/1523/files#diff-5b0ab37ac1fdf8ed411d298edc8c8e0fR190) we already return the saved value